include/device.h

```diff
@@ -293,6 +318,51 @@ void z_sys_device_do_config_level(s32_t level);
  */
 __syscall struct device *device_get_binding(const char *name);
 
+/** @brief Check if given extension is supported.
+ *
+ * @param dev		Device.
+ * @param api_mask	Bit mask identifiing an extension.
```

Comment:
I would use API ID number instead. 32 possibilities are not so much. Independently form fact how compatibility information is stored in drivers properties record.

Reply:
yes api id would allow extension beyond 32 bits and if 32 is too low at some point it can be internally extended without change of any API. I would like to keep it as bitmask internally to reduce overhead. Currently with device supporting only one api extension it is almost instant. Question is if we foresee devices with multple extensions (probably yes).